moviebot/user_modeling/user_model.py

```diff
@@ -0,0 +1,152 @@
+"""Class for user modeling."""
+
+
+import json
+import os
+from collections import defaultdict
+from typing import Dict, List
+
+from moviebot.database.database import DataBase
+
+
+class UserModel:
+    def __init__(self, user_id: str, historical_movie_choices_path: str = None):
+        """Initializes the user model.
+
+        The JSON file with historical movie choices is a dictionary with the
+        movie id as key and the list of choices as value.
+
+        Args:
+            user_id: User id.
+            historical_movie_choices_path: Path to the JSON file with historical
+              movie choices. Defaults to None.
+        """
+        self.user_id = user_id
+        self._movies_choices = defaultdict(str)
+
+        if historical_movie_choices_path and os.path.exists(
+            historical_movie_choices_path
+        ):
+            # Load historical movie choices
+            movie_choices = json.load(historical_movie_choices_path)
+            self._movies_choices.update(movie_choices)
+
+        self.tag_preferences = defaultdict(defaultdict(float))
```

Comment:
Default dict needs a callable as first argument.
```suggestion
        self.tag_preferences = defaultdict(lambda: defaultdict(float))
```

Reply:
Done.